node/src/main/scala/coop/rchain/node/grpc.scala

```diff
@@ -46,15 +48,29 @@ object GrpcServer {
       }.toFuture
   }
 
-  class DeployImpl[F[_]: Functor: MultiParentCasper: Futurable]
+  class DeployImpl[F[_]: Applicative: MultiParentCasper: Futurable]
       extends DeployServiceGrpc.DeployService {
-    def doDeploy(d: Deploy): Future[DeployServiceResponse] = {
-      val f = for {
-        _ <- MultiParentCasper[F].deploy(d)
-      } yield DeployServiceResponse(true)
+    override def doDeploy(d: DeployString): Future[DeployServiceResponse] =
+      InterpreterUtil.mkTerm(d.term) match {
+        case Right(term) =>
+          val deploy = Deploy(
+            user = d.user,
+            nonce = d.nonce,
+            term = Some(term),
+            sig = d.sig
+          )
+          val f = for {
+            _ <- MultiParentCasper[F].deploy(deploy)
+          } yield DeployServiceResponse(true, "Success!")
```

Comment:
@birchmd So how come this does't fall into the same issue of a stale `deploy` val ?

Reply:
Oh its not within a F. Nvm